lib/livebook_web/live/session_live.ex

```diff
@@ -1199,29 +1200,30 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, handle_operation(socket, operation)}
   end
 
-  def handle_info({:secret_created, %{origin: {:hub, _id}}}, socket) do
+  def handle_info({:secret_created, _secret}, socket) do
     {:noreply,
      socket
-     |> assign(saved_secrets: get_saved_secrets())
+     |> assign(saved_secrets: Hubs.get_secrets(socket.assigns.data_view.notebook_hub))
      |> put_flash(:info, "A new secret has been created on your Livebook Hub")}
   end
 
-  def handle_info({:secret_updated, %{origin: {:hub, _id}}}, socket) do
+  def handle_info({:secret_updated, _secret}, socket) do
     {:noreply,
      socket
-     |> assign(saved_secrets: get_saved_secrets())
+     |> assign(saved_secrets: Hubs.get_secrets(socket.assigns.data_view.notebook_hub))
      |> put_flash(:info, "An existing secret has been updated on your Livebook Hub")}
   end
 
-  def handle_info({:secret_deleted, %{origin: {:hub, _id}}}, socket) do
+  def handle_info({:secret_deleted, _secret}, socket) do
     {:noreply,
      socket
-     |> assign(saved_secrets: get_saved_secrets())
+     |> assign(saved_secrets: Hubs.get_secrets(socket.assigns.data_view.notebook_hub))
```

Comment:
```suggestion
     |> refetch_secrets()
```

and let's use `socket.private.data` :)